### PR TITLE
testcase: Adds a new test case for resource alicloud_nat_gateway

### PR DESCRIPTION
**Pull request by thisnihonglei**

### CODE REVIEW

File: alicloud/resource_alicloud_nat_gateway.go [hunk 1]
```diff
@@ -518,7 +518,7 @@ func resourceAlicloudNatGatewayDelete(d *schema.ResourceData, meta interface{})
 		return nil
 	})
 	if err != nil {
-		if IsExpectedErrors(err, []string{"INSTANCE_NOT_EXISTS", "IncorrectStatus.NatGateway", "InvalidNatGatewayId.NotFound", "InvalidRegionId.NotFound"}) {
+		if IsExpectedErrors(err, []string{"Forbidden.PrePaidNatGateway", "INSTANCE_NOT_EXISTS", "IncorrectStatus.NatGateway", "InvalidNatGatewayId.NotFound", "InvalidRegionId.NotFound"}) {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_nat_gateway.go:521]: 这个错误码不要增加，按照标准的预付费的删除逻辑写

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_nat_gateway.go (hunk 1 above)
Comment (thisnihonglei) [reply alicloud/resource_alicloud_nat_gateway.go:521]: 已去掉

Comment (xiaozhu36) [reply alicloud/resource_alicloud_nat_gateway.go]: 光去掉不行，delete API中要增加对预付费的处理

Comment (xiaozhu36) [reply alicloud/resource_alicloud_nat_gateway.go]: 并且在文档中要有说明

Comment (thisnihonglei) [reply alicloud/resource_alicloud_nat_gateway.go]: 已改

File: website/docs/r/nat_gateway.html.markdown [hunk 2]
```diff
@@ -18,7 +18,8 @@ If you want to add public IP, you can use resource 'alicloud_eip_association' to
 
 -> **NOTE:** From version 1.7.1, this resource has deprecated bandwidth packages.
 But, in order to manage stock bandwidth packages, version 1.13.0 re-support configuring 'bandwidth_packages'.
-    
+
+-> **NOTE:** `PrePaid` VPC NAT Gateway cannot be destroyed and provider will remove it from the state file.
```
Comment (xiaozhu36) [inline website/docs/r/nat_gateway.html.markdown:22]: 这个写清楚，当 payment_type 是 Subscription 或者 instance_charge_type 是prepaid 的时候是这样

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: website/docs/r/nat_gateway.html.markdown (hunk 2 above)
Comment (thisnihonglei) [reply website/docs/r/nat_gateway.html.markdown]: 已改

File: website/docs/r/nat_gateway.html.markdown [hunk 3]
```diff
@@ -18,7 +18,8 @@ If you want to add public IP, you can use resource 'alicloud_eip_association' to
 
 -> **NOTE:** From version 1.7.1, this resource has deprecated bandwidth packages.
 But, in order to manage stock bandwidth packages, version 1.13.0 re-support configuring 'bandwidth_packages'.
-    
+
+-> **NOTE:** The value of this attribute `payment_type` is `Subscription` or `Prepaid`, the VPC NAT Gateway cannot be destroyed and provider will remove it from the state file.
```
Comment (xiaozhu36) [inline website/docs/r/nat_gateway.html.markdown:22]: 改成如下的内容，并且把如下的内容放在 ## Attributes Reference 上方
“### Deleting `alicloud_nat_gateway` or  removing it from your configuration

The `alicloud_nat_gateway` resource allows you to manage `payment_type = "Subscription"` or `instance_charge_type = "Prepaid"` nat gateway, but Terraform cannot destroy it. 
Deleting the subscription resource or removing it from your configuration
will remove it from your statefile and management, but will not destroy the Nat Gateway.
You can resume managing the subscription nat gateway via the AlibabaCloud Console. “

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: website/docs/r/nat_gateway.html.markdown (hunk 3 above)
Comment (thisnihonglei) [reply website/docs/r/nat_gateway.html.markdown:22]: 已改

**Pull request merged**